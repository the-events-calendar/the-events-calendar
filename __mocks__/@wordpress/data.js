/**
 * External dependencies
 */
import { noop } from 'lodash';
import { Component } from 'react';

export const select = noop;
export const withSelect = () => ( component ) => component;
export const withDispatch = () => ( component ) => component;
