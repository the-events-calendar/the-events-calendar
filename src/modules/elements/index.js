export { default as Input } from './input';
export { default as TermsList } from './terms-list';
export { default as OrganizerForm } from './organizer-form';
export { default as VenueForm, toFields, toVenue } from './venue-form';
export { default as MetaGroup } from './meta-group';
export { default as SearchPosts } from './search-posts';
export { default as GoogleMap } from './google-map';
export { default as Dashboard } from './dashboard';
export { default as Month } from './month';
export { default as SearchOrCreate } from './search-or-create';
export { default as Loading } from './loading';
export { default as YearMonthForm } from './year-month-form';
export { default as Upsell } from './upsell';
export { default as TimeZone } from './timezone';
export { default as EditLink } from './edit-link';
