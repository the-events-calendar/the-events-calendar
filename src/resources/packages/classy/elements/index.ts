export * from './Classy';
export * from './fields';
