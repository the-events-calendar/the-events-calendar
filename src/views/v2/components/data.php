<?php
/**
 * View: Events Data Object.
 *
 * Override this template in your own theme by creating a file at:
 * [your-theme]/tribe/events/v2/components/data.php
 *
 * See more documentation about our views templating system.
 *
 * @link    http://evnt.is/1aiy
 *
 * @version 5.0.0
 * @var string                               $view_slug The slug of the view currently being rendered.
 * @var Tribe\Events\Views\V2\View_Interface $view      The View instance that is being rendered.
 *
 */

$data = $this->get_values();

/**
 * For very specific performance and security reasons we choose to not expose the full WP_Post object by default.
 * The following filter will allow by passing true, that the full WP_Post object gets exposed.
 *
 * @since 5.0.0
 *
 * @param boolean                              $should_expose_post_object If we should expose the events object or not.
 * @param array                                $data                      Data that will be exposed.
 * @param string                               $view_slug                 The slug of the view currently being rendered.
 * @param Tribe\Events\Views\V2\View_Interface $view                      The View instance that is being rendered.
 */
$should_expose_post_object = apply_filters( 'tribe_events_views_v2_view_data_should_expose_post_object', false, $data, $view_slug, $view );

if ( ! $should_expose_post_object ) {
	array_walk_recursive( $data, function ( &$value, $key ) {
		if ( $value instanceof WP_Post ) {
			$value = $value->ID;
		}
	} );
}

/**
 * Filters the data that will be printed for the View.
 *
 * @since 4.9.7
 *
 * @param array                                $data      The data that will be printed for the current View.
 * @param string                               $view_slug The slug of the view currently being rendered.
 * @param Tribe\Events\Views\V2\View_Interface $view      The View instance that is being rendered.
 */
$data = apply_filters( 'tribe_events_views_v2_view_data', $data, $view_slug, $view );

/**
 * Allows the filtering of the JSON options for testing and development purposes of the JSON encode options for the
 * view data JSON passed into the manager.
 *
 * @since 5.5.0
 *
 * @param int                                  $options   The Options passed into the json_encode.
 * @param string                               $view_slug The slug of the view currently being rendered.
 * @param Tribe\Events\Views\V2\View_Interface $view      The View instance that is being rendered.
 */
$json_options = apply_filters( 'tribe_events_views_v2_view_data_json_encode_options', 0 );
?>
<script data-js="tribe-events-view-data" type="application/json">
	<?php echo wp_json_encode( $data, $json_options ); ?>
</script>
