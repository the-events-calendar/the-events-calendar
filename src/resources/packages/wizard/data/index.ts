import { registerStore } from "@wordpress/data";
import {
	SETTINGS_STORE_KEY,
	SETTINGS_STORE_CONFIG
} from "./settings";

registerStore(SETTINGS_STORE_KEY, SETTINGS_STORE_CONFIG);

import {
	MODAL_STORE_KEY,
	MODAL_STORE_CONFIG
} from "./modal";

registerStore(MODAL_STORE_KEY, MODAL_STORE_CONFIG);

export {
	SETTINGS_STORE_KEY,
	MODAL_STORE_KEY
};
