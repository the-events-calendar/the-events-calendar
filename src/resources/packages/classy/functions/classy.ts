import {createRoot} from '@wordpress/element';
import {Classy} from "../elements";
import {getElement as getVisualEditorElement, toggleVisibility as toggleVisualEditorVisibility} from './visualEditor';

/**
 * Cached instance of the classy element.
 * Acts as a singleton to avoid creating multiple instances of the same element.
 *
 * @since TBD
 *
 * @type {HTMLElement|null}
 */
let classyElement: HTMLElement | null = null;

/**
 * Returns an existing Classy element or creates a new one if it doesn't exist.
 *
 * This function ensures that only one Classy element exists throughout the application
 * by caching the reference in the `classyElement` variable.
 *
 * @since TBD
 *
 * @param {Document|null} document - The document to use for element creation.
 *     If null, the global document will be used.
 * @returns {HTMLElement} The existing or newly created Classy element.
 */
export function getOrCreateElement(document: Document | null = null): HTMLElement {
	document = document ?? window.document;

	if (classyElement !== null) {
		return classyElement;
	}

	const element = document.createElement('div');
	element.id = 'tec-classy';
	element.classList.add('classy-root', 'classy-root--admin');
	classyElement = element;

	return element;
}

/**
 * Inserts the Classy element into the DOM after the visual editor.
 *
 * This function will get or create the Classy element and insert it into the DOM
 * immediately after the visual editor element. If the visual editor cannot be found,
 * the function will return false indicating failure.
 *
 * @since TBD
 *
 * @param {Document|null} document - The document to operate within.
 *     If null, the global document will be used.
 *
 * @returns {boolean} True if insertion was successful, false otherwise.
 */
export function insertElement(document: Document | null = null): boolean {
	document = document ?? window.document;
	const element = getOrCreateElement(document);
	const visualEditor = getVisualEditorElement(document);

	if (visualEditor === null) {
		return false;
	}

	visualEditor.parentNode.insertBefore(element, visualEditor.nextSibling);
	toggleVisualEditorVisibility(document);

	return true;
}

/**
 * Toggles the visibility of the Classy element.
 *
 * This function adds or removes the 'classy-root--hidden' class from the Classy element,
 * which should be styled with CSS to show or hide the element accordingly.
 * The element will be created if it doesn't already exist.
 *
 * @since TBD
 *
 * @param {Document|null} document - The document to operate within.
 *     If null, the global document will be used.
 *
 * @returns {void}
 */
export function toggleElementVisibility(document: Document | null = null): void {
	document = document ?? window.document;
	getOrCreateElement().classList.toggle('classy-root--hidden');
	toggleVisualEditorVisibility(document);
}

/**
 * Initializes the application by creating and rendering the Classy root element.
 *
 * This function ensures that the Classy element is either retrieved or created
 * and then initializes the React application by rendering the `Classy` component
 * into that element using React's `createRoot` API.
 *
 * @since TBD
 *
 * @param {Document|null} document - The document to use for element creation.
 *	 If null, the global document will be used.
 *
 * @returns {void}
 */
export function initApp(document: Document | null = null): void {
	document = document ?? window.document;
	const classyRoot = createRoot(getOrCreateElement(document));
	classyRoot.render(Classy());
}
