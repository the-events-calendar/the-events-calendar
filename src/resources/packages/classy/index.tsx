import {dispatch, select, subscribe} from "@wordpress/data";
import {localizedData} from './localized-data';
import {RichText} from '@wordpress/block-editor';
import {__} from "@wordpress/i18n";
import {registerPlugin} from '@wordpress/plugins';
import {createRoot} from '@wordpress/element';
import { SlotFillProvider, Slot, Fill } from '@wordpress/components';
import { addFilter, applyFilters } from '@wordpress/hooks';
import './style.pcss';

console.log('Classy!');

const {eventCategoryTaxonomyName} = localizedData;

function whenEditorIsReady(): Promise<void> {
	return new Promise((resolve: Function) => {
		const unsubscribe: Function = subscribe(() => {
			// This will trigger after the initial render blocking, before the window load event
			// This seems currently more reliable than using __unstableIsEditorReady
			if (select('core/editor').isCleanNewPost() || select('core/block-editor').getBlockCount() > 0) {
				unsubscribe()
				resolve()
			}
		})
	})
}

function createClassyElement(): Element {
	const classy = document.createElement('div');
	classy.id = 'tec-classy';
	classy.classList.add('classy-root', 'classy-root--admin');

	return classy;
}

function injectClassyElement(classyElement: Element): void {
	const visualEditor = document.querySelector('.editor-visual-editor');

	if (!visualEditor) {
		return;
	}

	visualEditor.parentNode.insertBefore(classyElement, visualEditor.nextSibling);
}

// Remove the category sidebar element.
dispatch('core/editor').removeEditorPanel(`taxonomy-panel-${eventCategoryTaxonomyName}`);

// Remove the post tag sidebar element.
dispatch('core/editor').removeEditorPanel('taxonomy-panel-post_tag');


function hideZoomOutButton(): void {
	// Remove the Zoom Out button. The only way is by its aria label.
	const zoomOutAriaLabel = __('Zoom Out');
	document.querySelectorAll(`.components-button[aria-label="${zoomOutAriaLabel}"]`)
		.forEach((el: Element) => el.style.display = 'none');
}

function hideInserterToggle() {
	// Remove the inserter toggle button.
	document.querySelectorAll('.editor-document-tools__inserter-toggle')
		.forEach((button: Element) => button.style.display = 'none');
}

function addEditorTools(onClick: (this: GlobalEventHandlers, ev: MouseEvent) => void): void {
	let editorToolsAdded = false;

	function EditorTools(): void {
		if (editorToolsAdded) {
			return;
		}

		const editorDocumentTools = document.querySelector('.editor-document-tools .editor-document-tools__left')

		if (editorDocumentTools) {
			const previewButton = document.createElement('button')
			previewButton.classList.add('tec-editor-tool', 'tec-editor-tool--preview', 'button');
			previewButton.type = 'button';
			previewButton.dataset.toolbarItem = 'true';
			previewButton.innerHTML = `<span class="dashicons dashicons-visibility"></span> ${__('Visual', 'the-events-calendar')}`;
			editorDocumentTools.append(previewButton);
			editorToolsAdded = true;
			previewButton.onclick = onClick;
		}

		return null;
	}

	registerPlugin('tec-editor-tools', {
		render: EditorTools
	});
}

const classyElement = createClassyElement();

whenEditorIsReady().then(() => {
	hideZoomOutButton();
	hideInserterToggle();

	addFilter(
		'classy.render',
		'additional-plugin',
		()=>(
			<Fill name="classy.fields">
				<div className="classy-input">
					<div className="classy-input__label">
						Fill
					</div>
					<div className="classy-input__input">
						<input type="text" placeholder="If you can read this, fill works!"/>
					</div>
				</div>
			</Fill>
		)
	);

	const classyRoot = createRoot(classyElement);

	classyRoot.render(
		<SlotFillProvider>

			{applyFilters('classy.render')}

			<div className="classy-container">
				<h3 className="classy-title">New Event</h3>

				<div className="classy-input">
					<div className="classy-input__label">
						Title
					</div>
					<div className="classy-input__input">
						<input type="text" placeholder="Your title"/>
					</div>
				</div>

				<div className="classy-input">
					<div className="classy-input__label">
						Description
					</div>
					<div className="classy-input__input">
						<RichText
							tagName="p"
							value=""
							allowedFormats={['core/bold', 'core/italic']}
							placeholder="Your event description"
						/>
					</div>
				</div>

				<Slot name="classy.fields" />

			</div>

		</SlotFillProvider>
	);

	injectClassyElement(classyElement);
	const toggleClassyVisibility = (): void => {
		console.log('classyElement', classyElement);
		classyElement.classList.toggle('classy-root--hidden');
	};
	addEditorTools(toggleClassyVisibility);
});
