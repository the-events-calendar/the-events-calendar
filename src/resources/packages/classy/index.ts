import { addAction, addFilter, didAction, doAction } from '@wordpress/hooks';
import renderFields from './functions/renderFields';
import { storeConfig } from './store';
import { STORE_NAME } from './constants';
import { getRegistry } from '@tec/common/classy/store';
import * as constants from './constants';
import { getNewStartEndDates, getAllDayNewDates, getMultiDayDates } from './functions/events';

/**
 * Hook on the Classy application initialization to add TEC store to the Classy registry.
 *
 * @since TBD
 *
 * @return {void} The ECP store is registered.
 */
const registerStore = (): void => {
	getRegistry().registerStore( STORE_NAME, storeConfig );

	/**
	 * Fires after the TEC store is registered and the TEC Classy application is initialized.
	 *
	 * @since TBD
	 *
	 * @return {void} The TEC store is registered.
	 */
	doAction( 'tec.classy.events.initialized' );
};

if ( didAction( 'tec.classy.initialized' ) ) {
	registerStore();
} else {
	addAction( 'tec.classy.initialized', 'tec.classy.events', registerStore );
}

// Hook on the Classy fields rendering logic to render the fields.
addFilter( 'tec.classy.render', 'tec.classy.events', renderFields );

// Export constants and functions to allow their reference in other packages.
export {
	constants,
	getNewStartEndDates,
	getAllDayNewDates,
	getMultiDayDates,
};
