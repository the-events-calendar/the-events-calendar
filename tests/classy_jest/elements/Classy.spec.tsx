import {
	afterEach,
	beforeAll,
	describe,
	expect,
	jest,
	test,
} from '@jest/globals';
import { Classy } from '../../../src/resources/packages/classy/elements';
import { render } from '@testing-library/react';
import { dispatch } from '@wordpress/data';
import {
	store,
	STORE_NAME,
} from '../../../src/resources/packages/classy/store';
import {
	registerMockStore,
	registerStoreIfNotRegistered,
	resetAllStores,
	unregisterStore,
} from '../__support__/store-mocks';

describe( 'Classy', () => {
	beforeAll( () => {
		registerStoreIfNotRegistered( STORE_NAME, store );
	} );

	afterEach( () => {
		jest.resetAllMocks();
		jest.restoreAllMocks();
		resetAllStores();
	} );

	describe( 'core/editor store available', () => {
		test( 'initial state render for new post', () => {
			registerMockStore( 'core/editor', {
				selectors: {
					getEditedPostAttribute: () => '',
					getEditedPostContent: () => '',
					getCurrentPostId: () => 2389,
				},
			} );

			const { container } = render( <Classy /> );

			expect( container ).toMatchSnapshot();
		} );

		test( 'initial state render with title for existing post', () => {
			registerMockStore( 'core/editor', {
				selectors: {
					getEditedPostAttribute( state, attribute: string ) {
						if ( attribute === 'title' ) {
							return 'Some Event';
						}

						if ( attribute === 'meta' ) {
							return {
								METADATA_EVENT_URL: 'https://example-event.com',
							};
						}

						return '';
					},
					getEditedPostContent: (): string => 'Lorem dolor',
					getCurrentPostId: () => 2389,
				},
			} );

			const { container } = render( <Classy /> );

			expect( container ).toMatchSnapshot();
		} );
	} );

	/* Skipping this while the alternate registry approach is worked out.
	describe( 'core/editor store not available', () => {
		beforeAll( () => {
			unregisterStore( 'core/editor' );
		} );

		test( 'initial state render for new post', () => {
			// @ts-ignore
			dispatch( STORE_NAME ).editPost( {
				title: '',
				content: '',
				currentPostId: 2389,
				meta: {},
			} );

			const { container } = render( <Classy /> );

			expect( container ).toMatchSnapshot();
		} );

		test( 'initial state render with title for existing post', () => {
			// @ts-ignore
			dispatch( STORE_NAME ).editPost( {
				title: 'Classy title',
				content: 'Classy content',
				currentPostId: 2389,
				meta: {
					METADATA_EVENT_URL: 'https://example-event.com',
				},
			} );

			const { container } = render( <Classy /> );

			expect( container ).toMatchSnapshot();
		} );
	} );
	*/
} );
