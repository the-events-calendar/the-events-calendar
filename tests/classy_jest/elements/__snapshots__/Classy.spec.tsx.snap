// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classy core/editor store available initial state render for new post 1`] = `
<div>
  <div
    class="classy-container"
  >
    <div
      class="classy-field classy-field--event-title"
    >
      <div
        class="classy-field__title"
      >
        <h3>
          Event Title
        </h3>
      </div>
      <div
        class="classy-field__inputs"
      >
        <div
          class="components-base-control components-input-control classy-field__control classy-field__control--input css-fothu4-Wrapper-boxSizingReset ej5x27r4"
        >
          <div
            class="components-base-control__field css-2txpkp-StyledField ej5x27r3"
          >
            <div
              class="components-flex components-input-base em5sgkm5 css-8kwvf6-PolymorphicDiv-Flex-base-ItemsColumn-Root e19lxcc00"
              data-wp-c16t="true"
              data-wp-component="InputBase"
            >
              <div
                class="components-input-control__container css-1x6d9pt-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"
              >
                <input
                  class="components-input-control__input css-19kifla-Input-dragStyles-fontSizeStyles-sizeStyles-customPaddings em5sgkm3"
                  id="inspector-input-control-0"
                  value=""
                />
                <div
                  aria-hidden="true"
                  class="components-input-control__backdrop css-13crio-BackdropUI-rtl em5sgkm6"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="classy-field classy-field--event-details"
    >
      <div
        class="classy-field__title"
      >
        <h3>
          Event Details
        </h3>
      </div>
      <div
        class="classy-field__inputs"
      >
        <div
          class="classy-field__input"
        >
          <div
            class="classy-field__input-title"
          >
            <h4>
              Description
            </h4>
          </div>
          <div
            class="classy-field__control classy-field__control--tinymce-editor"
          >
            <textarea
              class="classy-control-tinymce-editor wp-editor-area"
              id="classy-event-details-description-editor"
            />
          </div>
          <div
            class="classy-field__input-note"
          >
            Describe your event
          </div>
        </div>
        <div
          class="classy-field__input"
        >
          <div
            class="classy-field__input-title"
          >
            <h4>
              Featured Image
            </h4>
          </div>
          <div
            class="classy-field__control classy-field__control--featured-image"
          />
          <div
            class="classy-field__input-note"
          >
            We recommend a 16:9 aspect ratio for featured images.
          </div>
        </div>
        <div
          class="classy-field__input"
        >
          <div
            class="classy-field__input-title"
          >
            <h4>
              Event website
            </h4>
          </div>
          <div
            class="components-base-control components-input-control classy-field__control classy-field__control--input css-fothu4-Wrapper-boxSizingReset ej5x27r4"
          >
            <div
              class="components-base-control__field css-2txpkp-StyledField ej5x27r3"
            >
              <div
                class="components-flex components-input-base em5sgkm5 css-8kwvf6-PolymorphicDiv-Flex-base-ItemsColumn-Root e19lxcc00"
                data-wp-c16t="true"
                data-wp-component="InputBase"
              >
                <div
                  class="components-input-control__container css-1x6d9pt-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"
                >
                  <input
                    class="components-input-control__input css-19kifla-Input-dragStyles-fontSizeStyles-sizeStyles-customPaddings em5sgkm3"
                    id="inspector-input-control-1"
                    placeholder="www.example.com"
                    value=""
                  />
                  <div
                    aria-hidden="true"
                    class="components-input-control__backdrop css-13crio-BackdropUI-rtl em5sgkm6"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Classy core/editor store available initial state render with title for existing post 1`] = `
<div>
  <div
    class="classy-container"
  >
    <div
      class="classy-field classy-field--event-title"
    >
      <div
        class="classy-field__title"
      >
        <h3>
          Event Title
        </h3>
      </div>
      <div
        class="classy-field__inputs"
      >
        <div
          class="components-base-control components-input-control classy-field__control classy-field__control--input css-fothu4-Wrapper-boxSizingReset ej5x27r4"
        >
          <div
            class="components-base-control__field css-2txpkp-StyledField ej5x27r3"
          >
            <div
              class="components-flex components-input-base em5sgkm5 css-8kwvf6-PolymorphicDiv-Flex-base-ItemsColumn-Root e19lxcc00"
              data-wp-c16t="true"
              data-wp-component="InputBase"
            >
              <div
                class="components-input-control__container css-1x6d9pt-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"
              >
                <input
                  class="components-input-control__input css-19kifla-Input-dragStyles-fontSizeStyles-sizeStyles-customPaddings em5sgkm3"
                  id="inspector-input-control-2"
                  value="Some Event"
                />
                <div
                  aria-hidden="true"
                  class="components-input-control__backdrop css-13crio-BackdropUI-rtl em5sgkm6"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="classy-field classy-field--event-details"
    >
      <div
        class="classy-field__title"
      >
        <h3>
          Event Details
        </h3>
      </div>
      <div
        class="classy-field__inputs"
      >
        <div
          class="classy-field__input"
        >
          <div
            class="classy-field__input-title"
          >
            <h4>
              Description
            </h4>
          </div>
          <div
            class="classy-field__control classy-field__control--tinymce-editor"
          >
            <textarea
              class="classy-control-tinymce-editor wp-editor-area"
              id="classy-event-details-description-editor"
            >
              Lorem dolor
            </textarea>
          </div>
          <div
            class="classy-field__input-note"
          >
            Describe your event
          </div>
        </div>
        <div
          class="classy-field__input"
        >
          <div
            class="classy-field__input-title"
          >
            <h4>
              Featured Image
            </h4>
          </div>
          <div
            class="classy-field__control classy-field__control--featured-image"
          />
          <div
            class="classy-field__input-note"
          >
            We recommend a 16:9 aspect ratio for featured images.
          </div>
        </div>
        <div
          class="classy-field__input"
        >
          <div
            class="classy-field__input-title"
          >
            <h4>
              Event website
            </h4>
          </div>
          <div
            class="components-base-control components-input-control classy-field__control classy-field__control--input css-fothu4-Wrapper-boxSizingReset ej5x27r4"
          >
            <div
              class="components-base-control__field css-2txpkp-StyledField ej5x27r3"
            >
              <div
                class="components-flex components-input-base em5sgkm5 css-8kwvf6-PolymorphicDiv-Flex-base-ItemsColumn-Root e19lxcc00"
                data-wp-c16t="true"
                data-wp-component="InputBase"
              >
                <div
                  class="components-input-control__container css-1x6d9pt-Container-containerDisabledStyles-containerWidthStyles em5sgkm4"
                >
                  <input
                    class="components-input-control__input css-19kifla-Input-dragStyles-fontSizeStyles-sizeStyles-customPaddings em5sgkm3"
                    id="inspector-input-control-3"
                    placeholder="www.example.com"
                    value=""
                  />
                  <div
                    aria-hidden="true"
                    class="components-input-control__backdrop css-13crio-BackdropUI-rtl em5sgkm6"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
